fedimint-build/src/lib.rs

```diff
@@ -1,20 +1,36 @@
 use std::env;
-use std::fs::File;
-use std::process::Command;
+use std::process::{Command, Stdio};
+
+const FAKE_COMMIT_HASH: &str = "01234569afbe457afa1d2683a099c7af48a523c1";
 
 pub fn set_code_version() {
     if env::var_os("CODE_VERSION").is_none() {
-        let git_hash = if File::open("./../../.git/HEAD").is_ok() {
-            let output = Command::new("git")
-                .args(["rev-parse", "HEAD"])
-                .output()
-                .unwrap();
-            String::from_utf8(output.stdout).unwrap()
+        let output = match Command::new("git")
+            .args(["rev-parse", "HEAD"])
+            .stdout(Stdio::piped())
+            .stderr(Stdio::inherit())
+            .output()
+        {
+            Ok(o) => o,
+            Err(_) => {
```

Comment:
IMO this branch should be forbidden (fail the build), unless certain environment variable is set to allow it. We would set it in the CI that uses Nix and expect it, but otherwise we always expect the `git rev-parse HEAD` to just work. We might reconsider in the future when we start publishing these crates.